src/env.cc

```diff
@@ -675,6 +675,14 @@ Environment::Environment(IsolateData* isolate_data,
       thread_id_(thread_id.id == static_cast<uint64_t>(-1)
                      ? AllocateEnvironmentThreadId().id
                      : thread_id.id) {
+#ifdef NODE_V8_SHARED_RO_HEAP
+  if (isolate_data->worker_context() != nullptr) {
```

Comment:
Can it just be

```suggestion
  if (!is_main_thread()) {
```

? (techincally the same thing, but took me a while to realize that's what it means)

Reply:
Yup -- done in 492ede92978f51410022463795f3dd37a0c6ea19 🙂 Although I've added a DCHECK here to make sure that it still means that we're not dereferencing a `nullptr` when we access `...->worker_context()->env()`, probably not really necessary but DCHECKs are cheap :)